deposit-messaging/src/main/java/org/dataconservancy/pass/deposit/messaging/DepositApp.java

```diff
@@ -119,4 +133,54 @@ public static void main(String[] args) {
         app.run(args);
     }
 
-}
+    // When the Spring application starts and calls the Banner implementation, we take that opportunity
+    // to print the contents of the environment variables and resolved Spring properties using the
+    // provided Environment.
+    private static class DepositAppBanner implements Banner {
+
+        @Override
+        public void printBanner(Environment environment, Class<?> sourceClass, PrintStream out) {
+            LOG.info(">>>> Environment variable values:");
+
+            // Sort the variables by name and find the longest name
+            Map<String, String> vars = System.getenv();
+            List<String> keys = new ArrayList<>();
+            int maxLen = 0;
+            for (String varName : vars.keySet()) {
+                keys.add(varName);
+                if (varName.length() > maxLen) {
+                    maxLen = varName.length();
+                }
+            }
+            Collections.sort(keys);
+
+            // Print the variable names and values
+            for (String varName : keys) {
+                String nameString = StringUtils.rightPad(varName, maxLen);
+                LOG.info(">>>>   {} '{}'", nameString, vars.get(varName));
+            }
+
+            // Print out any resolved Spring property placeholders
+            boolean firstOne = true;
+            for (String varName : keys) {
+                String origValue = vars.get(varName);
+                String resolvedValue;
+                String errorMsg = "";
+                try {
+                    resolvedValue = environment.resolvePlaceholders(origValue);
```

Comment:
This is resolving a placeholder by the _value_ it originally had, rather than the _key_ it has.  I suspect this might be why I cannot get the application to print out any variables I've overrided.

Reply:
@birkland I was able to get it to work; there might be confusion how to test.  

After building `deposit-messaging`, I run the app just to make sure it works with a stock environment (e.g. `java -jar target/deposit-messaging-0.1.0-3.1-SNAPSHOT-exec.jar listen`).

Then I set an environment variable with a value of a property placeholder:
```
$ export DSPACE_HOST='${foo.bar}'
```

Then I set an environment variable with a name equivalent to the value of the property placeholder:
```
$ export FOO_BAR=localhost
```

Then when running the app I see:
```
...
[...] >>>> Environment variable values:
[...] >>>>   DSPACE_HOST                '${foo.bar}'
...
[...] >>>> Resolved Spring Environment property values:
[...] >>>>   DSPACE_HOST                'localhost' 
```

Printing out resolved placeholders with this secondary level of indirection may not be necessary, but it seems to cover all the bases.

Reply:
Hi Aaron,

Per Elliot, testing the resolution of Spring property placeholders involves setting two environment variables, one to identify the property placeholder and one to set its value.  Here is the example Elliot gave me, which I've tested with this code change:

(from Elliot) So for a test, you can make up a property and give it a value `PASS_FEDORA_PASSWORD=${my.password.placeholder}` and make sure that `my.password.placeholder` is in the `Environment` before attempting resolution (e.g. give the environment variable `MY_PASSWORD_PLACEHOLDER` a value, like `moo`)

Apparently, when Spring sees the environment variable MY_PASSWORD_PLACEHOLDER it creates a placeholder named my.password.placeholder that can be referenced with ${} to see its value (you probably know this).  Setting the second variable that uses the placeholder causes it to show up in the list I print.

    Ben